### PR TITLE
MSW mock server

### PR DESCRIPTION
**Pull request by Xoxol**
- implement integration with MSW mock server
- add mocks for user /login, /logout, /register
- add mocks for notices

### CODE REVIEW

File: src/mocks/notices/index.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    const noticeFromDB = data.notices.find((notice) => notice._id === id);
+    if (!noticeFromDB) {
+      return res(
+        ctx.status(404),
+      );
+    }
+
+    // deleting the notice in the DB
+    data.notices.splice(data.notices.indexOf(noticeFromDB), 1);
+    console.log(data.notices);
```
Comment (Aphelmion) [inline src/mocks/notices/index.js:146]: Is it necessary?

File: src/mocks/users/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import { API_HOST } from '@/constants';
+import { IS_AUTH } from '@/mocks/constants';
+
+import mockUtils from '../utils';
+import data from './mocks.json';
+
+export default [
+  // logging a user in
+  rest.post(`${API_HOST}/login`, (req, res, ctx) => {
+    console.log(req.body);
```
Comment (Aphelmion) [inline src/mocks/users/index.js:14]: Is it for debugging?

Comment (Xoxol) [reply src/mocks/users/index.js:14]: Debugging, will remove now

File: src/mocks/utils.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/*
+Probably it's overcomplicate mock server a bit and all these checks might be avoided
+ */
+
+const NOTICE_ALLOWED_FIELDS = ['authorized', 'title', 'description'];
+const NOTICE_REQUIRED_FIELDS = ['title', 'description'];
+
+const USER_ALLOWED_FIELDS = ['name', 'surname', 'egida_nick', 'phone', 'password', 'features'];
```
Comment (yana617) [inline src/mocks/utils.js:8]: User doesn't have a 'features' field

**Review by yana617 — APPROVED** [review]

**Review by Aphelmion — APPROVED** [review]

**Pull request merged**